lib/Runtime/Library/JavascriptArray.cpp

```diff
@@ -9154,7 +9154,7 @@ namespace Js
 
             while (count > 0)
             {
-                JS_REENTRANT(jsReentLock, BOOL hasItem = obj->HasItem(fromIndex));
+                JS_REENTRANT(jsReentLock, BOOL hasItem = JavascriptOperators::HasItem(obj, fromIndex));
```

Comment:
curious why not call OP_HasItem just like it is called above? Do you think we should make it consistent?

Reply:
OP_HasItem does some extra work because the instance and index are both Var. In this case we know that the object is a RecyclableObject and the index is a uint32, so we can bypass the extra work in the beginning of OP_HasItem. The case above is the slow path for numbers that don't fit in uint32.